pkg/runtime/runtime.go

```diff
@@ -935,17 +954,32 @@ func (a *DaprRuntime) initServiceDiscovery() error {
 }
 
 func (a *DaprRuntime) publishMessageHTTP(msg *pubsub.NewMessage) error {
+	subject := ""
+	var cloudEvent pubsub.CloudEventsEnvelope
+	err := a.json.Unmarshal(msg.Data, &cloudEvent)
+	if err == nil {
+		subject = cloudEvent.Subject
+	}
+
 	req := invokev1.NewInvokeMethodRequest(msg.Topic)
 	req.WithHTTPExtension(nethttp.MethodPost, "")
 	req.WithRawData(msg.Data, pubsub.ContentType)
 
-	// TODO Propagate Context
-	ctx := context.Background()
+	// subject contains the correlationID which is passed span context
+	sc, _ := diag.SpanContextFromString(subject)
+	ctx := diag.NewContext(context.Background(), sc)
+	spanName := fmt.Sprintf("PublishMessage: %s", msg.Topic)
```

Comment:
"SubscribeMessage: %s" does make sense.

Reply:
> "SubscribeMessage: %s" does make sense.

I earlier did that but Yaron suggested this name .

Reply:
What I suggested was for the subscribe method call, not for the publish method call. There is no subscription here.

Reply:
@yaron2 for user app point of view, isn't it subscribing message from kafka to user?

Reply:
Hence, since this is a publish http call to the app it should probably be PublishMessage.

Reply:
No, the subscription only happens once on init. 

Whatever happens after is publishing.

Reply:
So user will see trace like below

when user  app calls publish api, user will see `PublishEvent: xxx`
when user app gets the message, user will see `PublishMessage: xxx`

isn't it confusing to user?

Reply:
> No, the subscription only happens once on init.
> 
> Whatever happens after is publishing.

init will subscribe the topic. I understand. As a user perspective, how we can differenciate between `PublishEvent: xxx` and `PublishMessage: xxx` ?

Reply:
We can change it to DeliveredEvent, or not trace here at all since the app channel will be tracing anyway.

Reply:
`DeliveredEvent` is more clear.

Reply:
But won't this anyway get traced twice? Here and in app channel?

Reply:
> But won't this anyway get traced twice? Here and in app channel?

@yaron2 no tracing code in app channel in 0.7.0. We will do it in 0.8.0.

Reply:
What's this?

https://github.com/dapr/dapr/blob/749d7c25cd8705b2be1f93bbc454fa0b41dfcfd7/pkg/channel/http/http_channel.go#L97

Reply:
This is for metrics.

Reply:
You may think this is unclean, but as we use fasthttp, it doesn't have a concept for middleware for client side.

As I said, we will clean up the code in 0.8.0.

Reply:
Right. Hard to see on phone.

So if we put tracing in the app channel in 0.8.0 we will need to revisit this code in the runtime and remove it to not get duplicates.

Reply:
> Right. Hard to see on phone.
> 
> So if we put tracing in the app channel in 0.8.0 we will need to revisit this code in the runtime and remove it to not get duplicates.

We removed many duplicated span generations in 0.7.0. perf result may prove it between 0.6.0 and 0.7.0.

Even if we can see same code here and there in 0.7.0, we removed the actual duplicated unnecessary span generation in 0.7.0. 0.8.0 will have more cleaner code.

Reply:
> Hence, since this is a publish http call to the app it should probably be PublishMessage.

Yes. It is PublishMessage only currently. I earlier put it SubscribedMessage then later as per your suggestion I changed to PublishMessage.

Reply:
@yaron2 - so what is the final on name both in sendBindingEventToApp and publishMessageHTTP trace ?

Reply:
Looking at the amount of changes going in here, is it really needed in 0,7.1? IMO it can wait till 0.8.0, if this is addressing a regression, it should be targeted fix. Do we know what actually caused the regression and if tests don't exist for this?

Reply:
I have now changed it to DeliveredEvent.

Reply:
> We can change it to DeliveredEvent, or not trace here at all since the app channel will be tracing anyway.

@yaron2 , I have now changed it to DeliveredEvent.